.circleci/config.yml

```diff
@@ -3,110 +3,232 @@ version: 2.0
+      - run:
+          name: Install singularity
+          command: |
+            git clone https://github.com/sylabs/singularity.git $GOPATH/src/github.com/sylabs/singularity
+            cd $GOPATH/src/github.com/sylabs/singularity
             ./mconfig -p /usr/local
-            make -j$(nproc) -C ./builddir all
+            make -j$(nproc) -C ./builddir
             sudo make -C ./builddir install
       - run:
           name: Run tests
-          command: |-
-            cd $HOME/go/src/github.com/sylabs/singularity-cri
+          command: |
+            cd $GOPATH/src/github.com/sylabs/singularity-cri
             make test
+            cp cover.out $HOME
+      - store_artifacts:
+         path: ~/cover.out
+         destination: cover_profile.out
+
+  validation_test:
+    machine: true
+    steps:
+      - attach_workspace:
+          at: ~/go
+      - run:
+          name: Remount /run without noexec flag
+          command: |
+            sudo mount -o remount,exec /run
+      - run:
+          name: Setup environment
+          command: |
+            echo 'export GOPATH=$HOME/go' >> $BASH_ENV
+            echo 'export GOROOT=/usr/local/go' >> $BASH_ENV
+            echo 'export PATH=$PATH:$GOPATH/bin:$GOROOT/bin' >> $BASH_ENV
+            echo 'export PATH=$PATH:/usr/local/libexec/singularity/bin/' >> $BASH_ENV
+      - run:
+          name: Update go to 1.11
+          working_directory: /tmp
+          command: |
+            wget https://dl.google.com/go/go1.11.linux-amd64.tar.gz
+            sudo rm -rf /usr/local/go
+            sudo tar -C /usr/local -xzf go1.11.linux-amd64.tar.gz
+      - run:
+          name: Install dep tool
+          command: |
+            mkdir -p $GOPATH/bin
+            curl -s https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+      - run:
+          name: Install dev libraries
+          command: |
+            sudo apt-get update -y
+            sudo apt-get install -y build-essential libssl-dev uuid-dev squashfs-tools libgpgme11-dev
+            sudo apt-get install -y libseccomp2=2.2.3-2ubuntu1~ubuntu14.04.1
+            sudo apt-get install -y libseccomp-dev=2.2.3-2ubuntu1~ubuntu14.04.1
+            sudo apt-get install -y apparmor-profiles apparmor-utils socat
+      - run:
+          name: Install singularity
+          command: |
+            git clone https://github.com/sylabs/singularity.git $GOPATH/src/github.com/sylabs/singularity
+            cd $GOPATH/src/github.com/sylabs/singularity
+            ./mconfig -p /usr/local
+            make -j$(nproc) -C ./builddir
+            sudo make -C ./builddir install
+      - run:
+          name: Install singularity CRI
+          command: |
+            cd $GOPATH/src/github.com/sylabs/singularity-cri
+            make
+            sudo make install
+      - run:
+          name: Write CNI config for CRI
+          command: |
+            mkdir $HOME/net.d
+            cat > $HOME/net.d/00_bridge.conflist<< EOF
```

Comment:
Might be worth considering checking this in as a file in git and copying it to the right place here? May make maintenance a little easier.

Reply:
Where do you suggest storing that files? Under `.circleci` directory maybe?